### PR TITLE
Force Organization IDs to be lowercase

### PR DESCRIPTION
**Pull request by cragun47**
Signed-off-by: Brian Cragun <cragun@us.ibm.com>

Organization IDs  (Urls) are created from HL7 field content.    Anything that is not:   a-z A-Z 0-9 dash or period  is replaced by a period. 

This PR also lower-cases everything, adds missing tests, and checks for null values.

### CODE REVIEW

File: src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java [hunk 1]
```diff
@@ -208,12 +208,18 @@ public static String concatenateWithChar(Object input, String delimiterChar) {
...
+
+
     public static String formatAsId(Object input)
     {
-        // This replaces any special character (letters, numbers, dashes, or periods) with a period
-        String stringValue = Hl7DataHandlerUtil.getStringValue(input).trim();
-        stringValue = stringValue.replaceAll("[^a-zA-Z0-9\\-\\.]", ".");
-        return StringUtils.left(stringValue, 64);
+        if (input != null){
+            // This replaces any special character (letters, numbers, dashes, or periods) with a period
```
Comment (LisaWellman) [inline src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java:216]: the parenthetical should start with _anything other than_

Comment (LisaWellman) [reply src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java:216]: It also truncates to 64 characters, not sure why that is not mentioned

**Review by LisaWellman — APPROVED** [review]

File: src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java (hunk 1 above)
Comment (cragun47) [reply src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java:216]: good catch.
Fixed
